### ISSUE TITLE
Error : Input type (torch.FloatTensor) and weight type (torch.HalfTensor) should be the same 

### ISSUE THREAD

**Issue by niharpatel1999** [ultralytics/yolov5#101]
I trained my yolov5l model on Colab and updated the last.pt file to run inference on the local machine. But it shows the following error 
**RuntimeError: Input type (torch.FloatTensor) and weight type (torch.HalfTensor) should be the same**
while executing 
`python detect.py --weights weights/last.pt --img 416 --conf 0.4 --source ../test --device cpu`

**Comment by glenn-jocher** [comment]
@niharpatel1999 ok thanks for the bug report. I will try to reproduce.

**Comment by glenn-jocher** [comment]
I used our colab notebook to train a model for 3 epochs, then used last.pt to run inference. No problem at all, everything works fine.

<img width="1509" alt="Screen Shot 2020-06-16 at 10 26 41 AM" src="https://user-images.githubusercontent.com/26833433/84807300-e2f9fa80-afbb-11ea-8601-c6c5bd5a7ea8.png">

**Comment by niharpatel1999** [comment]
There is no issue if we run inference on colab .
But try downloading the last.pt file trained on GPU accelerated colab and
run inference on desktop than it would show error.

**Comment by glenn-jocher** [comment]
@niharpatel1999 ah, ok I see. I will try to reproduce your use case here.

**Comment by glenn-jocher** [comment]
@niharpatel1999 I repeated your steps, everything works fine for me. I don't see any problems. You may want to repeat your steps with the latest version of the repo, perhaps your bug has already been resolved.

You can also run models on cpu in colab like this by the way:
`python detect.py --device cpu`

**Comment by niharpatel1999** [comment]
Thanks.

**Comment by deeppatel4557** [comment]
@glenn-jocher, I am @niharpatel1999 teammate the weights tensor in our case is of the type torch.HalfTensor because we have trained it on GPU so there is an error of Input type torch.FloatTensor and weight type of torch.HalfTensor should be the same. we resolved that error just by changing one line of code in detect.py. Line number 23 has model.to(device).eval() to model.to(device).float().eval() by doing this we resolved the error by changing the type of weight tensor to float.It executed successfully on the CPU.Thanks for your support.
